app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaDetailsPresenter.kt

```diff
@@ -231,7 +231,11 @@ class MangaDetailsPresenter(
             return chapterList
         }
         getScrollType(chapterList)
-        return chapterSort.getChaptersSorted(chapterList)
+        val filteredChapterList = manga.filtered_scanlators?.let { filteredScanlatorString ->
+            val filteredScanlators = ChapterUtil.getScanlators(filteredScanlatorString)
+            return chapterList.filter { ChapterUtil.getScanlators(it.scanlator).any { group -> filteredScanlators.contains(group) } }
+        } ?: chapterList
+        return chapterSort.getChaptersSorted(filteredChapterList)
```

Comment:
thinking about this again, L234-237 should go inside of `getChaptersSorted` method instead. Since thats common and shared between the chapters in manga details and the chapters in the reader

I can make this change later if youre not feeling up for that

Reply:
I took a crack at it, not sure if i did it right, but it looks to work as intended when I tested